escape_roomba/context.py

```diff
@@ -0,0 +1,49 @@
+import asyncio
+import inspect
+
+
+class Context:
+    """Global data shared by bot subfunctions."""
+
+    def __init__(self, client):
+        self.client = client
+        self._event_listeners = {}
+
+    def add_listener(self, event_name, handler):
+        """Registers an async function for a Discord event (like 'on_message').
+        Multiple functions can be registered for the same event name."""
+
+        # discord.Client only supports one (static-ish) callback per event
+        # (discordpy.readthedocs.io/en/latest/api.html#discord.Client.event);
+        # discord.ext.commands.Bot/.Cog allow multiple subscribers, but also
+        # assume a specific UX, so we implement our own simple broadcasting.
```

Comment:
Nit: Is this comment meant for the whole of `add_listener`? Not immediately clear as I read this code top to bottom. If it is meant for `add_listener`, consider moving the comment just above/before the method starts? Also, I don't know what "static-ish" means in practical terms, so I find it kinda confusing -- either explain what you mean, or remove it?

Reply:
Moved up, removed the "static-ish" comment (it's not worth trying to explain).